torch/utils/collect_env.py

```diff
@@ -225,6 +230,139 @@ def get_nvidia_smi():
     return smi
 
 
+def get_pkg_version(run_lambda, pkg):
+    ret = ""
+    index = -1
+    if get_platform() == "linux":
+        mgr_name = ""
+        if mgr_name == "":
+            rc, _, _ = run("which dpkg")
+            if rc == 0:
+                mgr_name = "dpkg"
+        if mgr_name == "":
+            rc, _, _ = run("which dnf")
+            if rc == 0:
+                mgr_name = "dnf"
+        if mgr_name == "":
+            rc, _, _ = run("which yum")
+            if rc == 0:
+                mgr_name = "yum"
+        if mgr_name == "":
+            rc, _, _ = run("which zypper")
+            if rc == 0:
+                mgr_name = "zypper"
```

Comment:
Please Avoid code duplication, use loops
```suggestion
        for mgr_name in ["dpkg", "dnf", "yum", "zypper", ""]:
            if mgr_name == "":
                continue
            rc, _, _ = run(f"which {mgr_name}")
            if rc == 0:
                break
```